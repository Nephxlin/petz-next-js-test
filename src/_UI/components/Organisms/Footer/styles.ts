import styled from "styled-components";

export const Footer = styled.footer`
  background-color: #1D1D1D;
  color: #FFFFFF;
  text-align: center;
  padding-top: 24px;
  padding-bottom: 31px;
`