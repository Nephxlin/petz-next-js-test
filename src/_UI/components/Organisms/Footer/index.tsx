import * as S from './styles'

const Footer = () => {
  return (
    <S.Footer data-testid="footer-component">
      <p>Todas as marcas e ilustrações utilizadas são de seus resepctivos donos.</p>
    </S.Footer>
  )
}

export default Footer