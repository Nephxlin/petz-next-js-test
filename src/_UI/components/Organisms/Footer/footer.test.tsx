import 'jest-styled-components';
import { screen, render } from '@testing-library/react';
import Footer from './index';

describe('Footer component', () => {
  it('renders Footer', () => {
    render(
      <Footer />
    );
    const footer = screen.getByTestId('footer-component')
    expect(footer).toBeInTheDocument();
    expect(footer).toHaveStyleRule('background-color', '#1D1D1D');
    expect(footer).toHaveStyleRule('text-align', 'center');
    expect(footer).toMatchSnapshot();
  });
});
