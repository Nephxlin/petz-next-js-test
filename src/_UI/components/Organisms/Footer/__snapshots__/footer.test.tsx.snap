// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders Footer 1`] = `
.c0 {
  background-color: #1D1D1D;
  color: #FFFFFF;
  text-align: center;
  padding-top: 24px;
  padding-bottom: 31px;
}

<footer
  class="c0"
  data-testid="footer-component"
>
  <p>
    Todas as marcas e ilustrações utilizadas são de seus resepctivos donos.
  </p>
</footer>
`;
